### PR TITLE
Get rid of the legacy mongodb integration

### PR DESCRIPTION
**Pull request by allmazz**
### Changelog category (leave one):
- Backward Incompatible Change

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
The legacy MongoDB integration based on the Poco driver has been removed. Server setting `use_legacy_mongodb_integration` is obsolete and has no effect anymore.

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit fcaa058c9271658a9e0bf180eaa2d6f0b298746d with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/71997/fcaa058c9271658a9e0bf180eaa2d6f0b298746d/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/71997/fcaa058c9271658a9e0bf180eaa2d6f0b298746d/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

File: src/Core/ServerSettings.cpp [hunk 1]
```diff
@@ -194,7 +194,6 @@ namespace DB
     DECLARE(UInt64, parts_kill_delay_period_random_add, 10, "Add uniformly distributed value from 0 to x seconds to kill_delay_period to avoid thundering herd effect and subsequent DoS of ZooKeeper in case of very large number of tables. Only available in ClickHouse Cloud", 0) \
     DECLARE(UInt64, parts_killer_pool_size, 128, "Threads for cleanup of shared merge tree outdated threads. Only available in ClickHouse Cloud", 0) \
     DECLARE(UInt64, keeper_multiread_batch_size, 10'000, "Maximum size of batch for MultiRead request to [Zoo]Keeper that support batching. If set to 0, batching is disabled. Available only in ClickHouse Cloud.", 0) \
-    DECLARE(Bool, use_legacy_mongodb_integration, true, "Use the legacy MongoDB integration implementation. Note: it's highly recommended to set this option to false, since legacy implementation will be removed in the future. Please submit any issues you encounter with the new implementation.", 0) \
```
Comment (nikitamikhaylov) [inline src/Core/ServerSettings.cpp:197]: We should never remove settings. Instead, please make it obsolete.

**Comment by rschu1ze** [comment]
ClickHouse Stateless tests (ubsan) [2/2]: https://github.com/ClickHouse/ClickHouse/issues/72008

File: src/Core/ServerSettings.cpp [hunk 2]
```diff
@@ -201,7 +201,7 @@ namespace DB
...
+    DECLARE(Bool, use_legacy_mongodb_integration, true, "Obsolete, has no effect", 0) \
```
Comment (Algunenano) [inline src/Core/ServerSettings.cpp:204]: Changing the description to say it's obsolete does nothing. You need to change the flags themselves

**Pull request merged**